### DISCUSSION TITLE
Can YARA handle searching on streaming data?

### DISCUSSION THREAD

**Discussion by sdtckr** [VirusTotal/yara#2116] [Q&A] [answered]
I’m exploring YARA for a project where we need to perform pattern matching on streaming data, which arrives in parts rather than as a single file. I know Hyperscan has a streaming mode that allows it to keep track of state across multiple data chunks, but I’m unsure if YARA has a similar feature.
Does YARA natively support any kind of stream-based searching, or is it strictly for complete data files?

**Comment by plusvic** [comment, marked as answer]
Hi @sdtckr, no, YARA is a block-based scanner and it doesn't offer support for stream scanning.
